File: app/src/main/java/br/com/medeve/Helpers/IntentHelper.java
package br.com.medeve.Helpers;

import android.app.Activity;
import android.content.Intent;

import br.com.medeve.Interfaces.IIntent;

public class IntentHelper implements IIntent {

    private static IntentHelper intentHelper;

    public static synchronized IntentHelper getInstance() {
        if (intentHelper == null) {
            intentHelper = new IntentHelper();
        }
        return intentHelper;
    }

    @Override
    public void intentWithFinish(Activity activity,Class clazz) {
        Intent intent = new Intent(activity,clazz);
        activity.startActivity(intent);
        activity.finish();
    }

    @Override
    public void intentWithOutFinish(Activity activity, Class clazz) {
        Intent intent = new Intent(activity, clazz);
        activity.startActivity(intent);
    }

    @Override
    public void intentWithFlags(Activity activity, Class clazz) {
        Intent intent = new Intent(activity, clazz);
        activity.startActivity(intent);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
    }
}

File: app/src/main/java/br/com/medeve/Helpers/IntentHelper.kt
package br.com.medeve.Helpers

import android.app.Activity
import android.content.Intent
import br.com.medeve.Interfaces.IIntent

object IntentHelper : IIntent {
    override fun intentWithFinish(activity: Activity?, clazz: Class<*>?) {
        val intent = Intent(activity, clazz)
        activity!!.startActivity(intent)
        activity.finish()
    }

    override fun intentWithOutFinish(activity: Activity?, clazz: Class<*>?) {
        val intent = Intent(activity, clazz)
        activity!!.startActivity(intent)
    }

    override fun intentWithFlags(activity: Activity?, clazz: Class<*>?) {
        val intent = Intent(activity, clazz)
        activity!!.startActivity(intent)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
    }

}